crates/common/mqtt_channel/src/config.rs

```diff
@@ -136,6 +142,14 @@ impl Config {
         }
     }
 
+    /// Set the initial message flag
+    pub fn with_initial_message(self, initial_message: Option<fn() -> Message>) -> Self {
+        Self {
+            initial_message,
+            ..self
+        }
+    }
```

Comment:
I would also consider to use an [`Fn`](https://doc.rust-lang.org/std/ops/trait.Fn.html) and not a simple [`fn`](https://doc.rust-lang.org/std/primitive.fn.html). The difference is [subtle](https://users.rust-lang.org/t/fn-vs-fn-in-type-signatures/26108) but this will give the ability to use closures instead of plain functions (e.g `with_initial_message(||-> get_health_status_message("tedge-agent")`)